### ISSUE TITLE
Handle Numba/LLVM delays for new Python versions

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/PlasmaPy#1342]
It usually takes a ∼1–2 months for LLVM and hence Numba to become compatible with new versions of Python.  At present, this is blocking us from being compatible with Python 3.10.  This is one of the things that would block us from being able to test PlasmaPy on the development version of Python or alpha/beta/rc releases (see also discussion in #1215).  

One way to get around this would be to have substitute decorators that could be imported when Numba cannot be imported.  Essentially we'd be creating a blank interface within PlasmaPy that could be aliased to `numba`.  

One possibility would be to include in our main `__init__.py` a try/except block that would `import numba as _numba`, and if that raises an `ImportError`, then we'd `import plasmapy.utils.fake_numba as _numba`.  In the rest of the code we'd then do `from plasmapy import _numba` and decorate things like `@_numba.jit`.  I'm not sure which namespace would be best to use.

My preference would be to issue a warning that Numba can't be imported.  We could do this either when PlasmaPy is first imported, or when a function is called that is supposed to have been Numbafied.  I think I prefer the latter.  It'd probably be helpful to include in the warning a mention that for newly released versions of Python, it might take a month or two for Numba to become compatible with it, since that would be really useful context.

**Comment by StanczakDominik** [comment]
On some further thought... This will work for `numba.jit`, which wraps the decorated function without changing its functionality, but won't work for [`numba.vectorize`, which turns the function into a ufunc](https://numba.pydata.org/numba-doc/dev/user/vectorize.html). @rocco8773 mentioned using `numba.guvectorize` - same thing applies there. We do actually need numba there for the functionality, not just for the JIT. 

Consider this snippet:
```python
import numba
import numpy as np
from numba import float64

@numba.jit
def the_answer():
    return 42

print(the_answer())

@numba.vectorize([float64(float64)])
def vectorized_answer(x):
    return 42.0
args = np.arange(10)
answers = vectorized_answer(args)
print(answers)
assert answers.shape == (10,)
```

and run it as `python numba_test.py` and `NUMBA_DISABLE_JIT=1 python numba_test.py`. In the latter case, numba doesn't JIT anything; but it still returns a vector function that conserves the shape of the argument.

Looking at how https://github.com/poliastro/poliastro handles that problem... [they just run a hard numba dependency nowadays.](https://docs.poliastro.space/en/stable/installation.html) This is also sane for another reason: for some algorithm implementation, the lack of numba means it actually isn't going to run in reasonable time at all.

To sum up: I think going hard on the numba dependency would be just as viable and have fewer side effects.
